libraries/x86/src/x86_tcp.c

```diff
@@ -0,0 +1,124 @@
+#include "x86_tcp.h"
+#include <arpa/inet.h>
+#include <errno.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <pthread.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#define X86_TCP_PORT 4000
```

Comment:
If we're using a fixed address, we should add some way to distinguish between devices. It might be easier to use a random address and parse it from the output of the program.